src/Elastic.Apm/Logging/IApmLoggingExtensions.cs

```diff
@@ -1,60 +1,99 @@
+		/// otherwise it returns null.
+		/// By using the return value with `?.` you can avoid executing code that is not necessary to execute
+		/// in case the log won't be printed because the loglevel would not allow it.
+		/// </summary>
+		/// <param name="logger">The logger instance you want to log with</param>
+		/// <returns>Either a MaybeLogger or null</returns>
+		internal static MaybeLogger? Warning(this IApmLogger logger) => IfLevel(logger, LogLevel.Warning);
 
-		public static void LogDebugException(this IApmLogger logger, Exception exception) =>
-			logger?.LogDebug(exception, "{ExceptionName}: {ExceptionMessage}", exception?.GetType().Name, exception?.Message);
+		/// <summary>
+		/// If the logger has a loglevel, which is higher than Info then it returns a MaybeLogger instance,
+		/// otherwise it returns null.
+		/// By using the return value with `?.` you can avoid executing code that is not necessary to execute
+		/// in case the log won't be printed because the loglevel would not allow it.
+		/// </summary>
+		/// <param name="logger">The logger instance you want to log with</param>
+		/// <returns>Either a MaybeLogger or null</returns>
+		internal static MaybeLogger? Info(this IApmLogger logger) => IfLevel(logger, LogLevel.Information);
 
-		public static void LogErrorException(this IApmLogger logger, Exception exception,
-			[System.Runtime.CompilerServices.CallerMemberName]
-			string method = "",
-			[System.Runtime.CompilerServices.CallerFilePath]
-			string filepath = "",
-			[System.Runtime.CompilerServices.CallerLineNumber]
-			int lineNumber = 0
-		)
+		internal readonly struct MaybeLogger
 		{
-			var file = string.IsNullOrEmpty(filepath) ? string.Empty : new FileInfo(filepath).Name;
-			logger?.LogError(exception, "{ExceptionName} in {Method} ({File}:{LineNumber}): {ExceptionMessage}", exception?.GetType().Name, method, file, lineNumber, exception?.Message);
-		}
+			private readonly IApmLogger _logger;
+			private readonly LogLevel _level;
 
-		public static ScopedLogger Scoped(this IApmLogger logger, string scope) =>
-			 new ScopedLogger((logger is ScopedLogger s ? s.Logger : logger), scope);
+			public MaybeLogger(IApmLogger logger, LogLevel level) => (_logger, _level) = (logger, level);
 
-		private static void DoLog(this IApmLogger logger, LogLevel level, string message, Exception e, object[] args)
-		{
-			var formatter = logger is ScopedLogger sl
-				? sl.GetOrAddFormatter(message, args.Length)
-				: Formatters.GetOrAdd(message, s => new LogValuesFormatter(s));
+			public void Log(string message, params object[] args) => _logger.DoLog(_level, message, null, args);
 
-			var logValues = formatter.GetState(args);
-			logger?.Log(level, logValues, e, (s, _) => formatter.Format(args));
+			public void LogException(Exception exception, string message, params object[] args) =>
+				_logger.DoLog(_level, $"Exception: {{ExceptionType}}, Message: {{ExceptionMsg}} \n{message}", exception,
```

Comment:
We add the details of the exception to the log message here. Optionally we can move this to `ConsoleLogger`. My intention with this is that I think we should have the `Exception` also in a textual form and print that to the user. It's nice to have it in a structured form for sophisticated logging frameworks, but for a simple case like `ConsoleLogger` we should also give that info within the text that we print.

Reply:
https://github.com/aspnet/Extensions/blob/master/src/Logging/Logging.Console/src/ConsoleLogger.cs

Adapters typically take care of this already when floating exception. I'm personally +1 to move this to our consolelogger as well

Reply:
I see. Agree, let's do it like `Microsoft.Extensions.Logging.Console.ConsoleLogger` - moved this part into our `ConsoleLogger`.